src/error.rs

```diff
@@ -41,6 +41,58 @@ impl Display for InvalidProbeId {
     }
 }
 
+/// Error that indicates an invalid wall clock time was detected.
+///
+/// Wall clock time must not be greater than Nanoseconds::MAX
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct InvalidWallClockTime;
+
+#[cfg(feature = "std")]
+impl std::error::Error for InvalidWallClockTime {}
+
+#[cfg(feature = "std")]
+impl Display for InvalidWallClockTime {
```

Comment:
Looks like this impl of `Display` is compatible with the features available in `core::fmt`, probably could ease up on the std-flagging here with a std->core module rename.